### ISSUE TITLE
SlevomatCodingStandard.Functions.TrailingCommaInCall.MissingTrailingComma false positive for set amount of parameters

### ISSUE THREAD

**Issue by patrickkusebauch** [slevomat/coding-standard#768]
Hello,
I found an issue, where this sniff causes false positives in the case when I call a function that is not variadic nor does it have unused/default parameters. An example might be better: 

 - For a call to a variadic function, it works correctly:
```php
function test_1(...$many_parameters) {
 //whatever, not important
}

test_1(
'a',
'b'
)
``` 
It will raise an error.

 - For a function with default parameter values, the behavior might be debatable, but it can be argued it works correctly:
```php
function test_2($a, $b, $c = null) {
 //whatever, not important
}

test_2(
'a',
'b'
)
``` 
It will raise an error.

 - For a function where all parameters have been supplied, there is no reason to require a trailing comma:
```php
function test_3($a, $b) {
 //whatever, not important
}

test_3(
'a',
'b'
)
``` 
It will raise an error. I have a reason it should not. The trailing comma here implies I can supply more parameters to this function, but it is not possible and the trailing comma, therefore, confuses the developer.

**Comment by kukulich** [comment]
Trailing comma is required because of better future diff.

**Comment by patrickkusebauch** [comment]
@kukulich I have read the RFC and the reasoning behind the addition of trailing comma. I have properly noted that in the first 2 cases the sniff checks for the desired behavior. 

In the last case, the should not be a comma implying the possibility of adding a 3rd parameter, when the function accepts only 2 parameters.

There is no situation when I might want to add a new line with one more parameters since the function does not accept any more parameters, therefore the comma is not needed for better future diff.

**Comment by kukulich** [comment]
@patrickkusebauch The function can have more parameters in future.

**Comment by patrickkusebauch** [comment]
@kukulich One last note on this, I promise, since you seem convinced in the solution.

For the argument:
> The function can have more parameters in future.

The PHP developers have decided trailing comma [should not be included](https://wiki.php.net/rfc/list-syntax-trailing-commas) when the function might have more parameters in the future. The inclusion of trailing comma in **function declaration** was the explicit reason why that part of RFC did not pass as was noted in the subsequent [RFC](https://wiki.php.net/rfc/trailing-comma-function-calls). 

Therefore in my mind, it has been decided trailing comma as an argument for the function possibly having more parameters in the future has been denied in the function declaration and therefore should not be allowed in function invocation.

This sniff, therefore, by requiring the trailing comma creates internal inconsistency where you are saying you must have the comma in the invocation but must not have it in the declaration.
